osu.Game/Screens/Select/BeatmapPlaylistItem.cs

```diff
@@ -0,0 +1,311 @@
+
+            isHovered = true;
+
+            showHoverElements(true);
+            return true;
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            isHovered = false;
+
+            if (isDragged)
+                return;
+
+            showHoverElements(false);
+        }
+
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            // We manually track dragging status as to not capture dragging events (so we don't interfere with the scrolling behaviour of our parent)
+            isDragged = true;
+            return false;
+        }
+
+        protected override bool OnMouseUp(MouseUpEvent e)
+        {
+            isDragged = false;
+
+            if (!isHovered)
+                showHoverElements(false);
+
+            return base.OnMouseUp(e);
+        }
+
+        protected override bool OnMouseMove(MouseMoveEvent e)
+        {
+            // This is to show the current item's buttons after having dragged a different item and landing here (i.e. the OnHover was prevented from being fired)
+            if (!isHovered && !e.IsPressed(MouseButton.Left))
+                showHoverElements(true);
+
+            return base.OnMouseMove(e);
+        }
+
+        private void showHoverElements(bool show)
+        {
+            if (show)
+            {
+                removeButton.Show();
+                dragHandle.Show();
+            }
+            else
+            {
+                removeButton.Hide();
+                dragHandle.Hide();
+            }
+        }
+
+        private class DragHandle : SpriteIcon
+        {
+            public DragHandle()
+            {
+                Anchor = Anchor.CentreLeft;
+                Origin = Anchor.CentreLeft;
+                Size = new Vector2(12);
+                Icon = FontAwesome.fa_bars;
+                Alpha = 0;
+                Margin = new MarginPadding { Left = 5, Top = 2 };
+            }
+
+            public override void Show()
+            {
+                this.FadeIn(fade_duration);
+            }
+
+            public override void Hide()
+            {
+                this.FadeOut(fade_duration);
+            }
+        }
+
+        private class RemoveButton : OsuClickableContainer
+        {
+            public RemoveButton()
+            {
+                Alpha = 0;
+                Child = new SpriteIcon
+                {
+                    Colour = Color4.White,
+                    Icon = FontAwesome.fa_minus_square,
+                    Size = new Vector2(14),
+                };
+            }
+
+            public override void Show()
+            {
+                this.FadeIn(fade_duration);
+            }
+
+            public override void Hide()
+            {
+                this.FadeOut(fade_duration);
+            }
+
+            protected override bool OnMouseDown(MouseDownEvent e)
+            {
+                Content.ScaleTo(0.75f, 2000, Easing.OutQuint);
```

Comment:
Rather than using `Content` here I'd store the icon to a private and adjust there, just for clarity. We generally don't use `Content.` if we don't own the content.